import/index_java/src/org/vufind/index/ConfigManager.java

```diff
@@ -210,55 +225,48 @@ public Ini loadConfigFile(String filename)
                 retVal.put(key, overrideSection.get(key));
             }
         }
+
         return retVal;
     }
 
     /**
-     * Get a setting from a VuFind configuration file.
+     * @deprecated
+     */
+    public Map<String, String> getSanitizedConfigSection(String filename, String section)
+    {
+        return getConfigSection(filename, section);
+    }
+
+    /**
+     * Get a setting from a VuFind configuration file and sanitize the value.
      * @param filename configuration file name
      * @param section section name within the file
      * @param setting setting name within the section
      */
     public String getConfigSetting(String filename, String section, String setting)
     {
-        String retVal = null;
-
-        // Grab the ini file.
-        Ini ini = loadConfigFile(filename);
-
-        // Check to see if we need to worry about an override file:
-        String override = ini.get("Extra_Config", "local_overrides");
-        if (override != null) {
-            Ini overrideIni = loadConfigFile(override);
-            retVal = overrideIni.get(section, setting);
-            if (retVal != null) {
-                return sanitizeConfigSetting(retVal);
-            }
-        }
-
-        // Try to find the requested setting:
-        retVal = ini.get(section, setting);
+        String retVal = getRawConfigSetting(filename, section, setting);
+        return retVal == null ? retVal : sanitizeConfigSetting(retVal);
+    }
 
-        //  No setting?  Check for a parent configuration:
-        while (retVal == null) {
-            String parent = ini.get("Parent_Config", "path");
-            if (parent !=  null) {
-                try {
-                    ini.load(new FileReader(new File(parent)));
-                } catch (Throwable e) {
-                    dieWithError(
-                        "Unable to access " + parent
-                        + " (" + e.getMessage() + ")"
-                    );
-                }
-                retVal = ini.get(section, setting);
-            } else {
-                break;
-            }
-        }
+    /**
+     * Get a setting from a VuFind configuration file.
+     * @param filename configuration file name
+     * @param section section name within the file
+     * @param setting setting name within the section
+     */
+    public String getRawConfigSetting(String filename, String section, String setting)
+    {
+        Map<String, String> sectionMap = getRawConfigSection(filename, section);
+        return sectionMap == null ? null : sectionMap.get(setting);
+    }
 
-        // Return the processed setting:
-        return retVal == null ? null : sanitizeConfigSetting(retVal);
+    /**
+     * @deprecated
```

Comment:
Would it be better to use the `@Deprecated` Java annotation outside of the comment? Or do we possibly need both? I'm not sure of the best convention, but while I was testing, I noticed that the compiler was throwing warnings about having deprecated methods without the annotation. Any thoughts?

Reply:
I'm no real Java expert so i can only guess, and even SolrMarc doesn't seem to be consistent in this case...

Most IDEs support JavaDoc and e.g. in Netbeans the deprecated functions will be highlighted in a special way. I think the only advantage of using the `@Deprecated` annotation might be to display warnings when compiling or executing so that people would see it more often when migrating to a new version - and that actually could be worth doing it. Some people even recommend to do both, to mark as `@Deprecated` and provide additional information in the JavaDoc as well (in Java 9 `@Deprecated` received additional attributes like "since" and "forRemoval"), but of course many people still rely on 8 and also we could add additional info in the JavaDoc like e.g. information about the successor-function. 

What do you think?

Reply:
I like what you've done; I think this is the most useful approach. I have approved the PR and will merge soon -- just need to find time to write up the changelog notes. :-)